src/main/java/sirius/biz/analytics/metrics/ComputeParameters.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.analytics.metrics;
+
+import sirius.db.mixing.BaseEntity;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+/**
+ * Contains the parameters passed to {@link DailyMetricComputer#compute(ComputeParameters)},
+ * {@link MonthlyMetricComputer#compute(ComputeParameters)} and
+ * {@link MonthlyLargeMetricComputer#compute(ComputeParameters)}.
+ *
+ * @param date                           the date for which the computation should be performed
+ * @param startOfPeriod                  the start of the period as {@link LocalDateTime}
+ * @param endOfPeriod                    the end of the period as {@link LocalDateTime}
+ * @param periodOutsideOfCurrentInterest <b>true</b> if the computation is performed for a past or future period
```

Comment:
can't this be a method and calculated _live_ based on the top 3 parameters?

Reply:
Could be, yes … but depending on the scheduling, there are many cases, and it would be an even bigger change … I would prefer to leave it like this